Error [ERR_REQUIRE_ESM]: Must use import to load ES Module @preact\preset-vite when scaffolding vanilla preact

When using `npm init vite` with the `preact` framework and `preact` (vanilla js) variant, I get the following error when starting the development server after npm install:

```
error when starting dev server:
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: E:\Javascript\starter-vite-preact\node_modules\@preact\preset-vite\dist\index.js
require() of ES modules is not supported.
require() of E:\Javascript\starter-vite-preact\node_modules\@preact\preset-vite\dist\index.js from E:\Javascript\starter-vite-preact\vite.config.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from E:\Random42\Javascript\starter-vite-preact\node_modules\@preact\preset-vite\package.json.

    at Module._extensions..js (internal/modules/cjs/loader.js:1080:13)
    at Object.require.extensions.<computed> [as .js] (E:\Javascript\starter-vite-preact\node_modules\vite\dist\node\chunks\dep-36bf480c.js:77286:13)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (E:\Random42\Javascript\starter-vite-preact\vite.config.js:30:37)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.require.extensions.<computed> [as .js] (E:\Javascript\starter-vite-preact\node_modules\vite\dist\node\chunks\dep-36bf480c.js:77283:20)
    at Module.load (internal/modules/cjs/loader.js:928:32)
```

Deps versions are
```
    "preact": "^10.5.13"
    "@preact/preset-vite": "^2.0.0",
    "vite": "^2.5.4"
```

Same here.
The error goes away after adding `"type": "module"` in package.json, however VSCode started to compain now (*Cannot find module '@preact/preset-vite' or its corresponding type declarations.*).

I didn't understand why it's not been set in the templates (reference: https://github.com/vitejs/vite/pull/4293)

--- a/package.json
+++ b/package.json
@@ -2,13 +2,21 @@
 	"name": "@preact/preset-vite",
 	"version": "2.1.1",
 	"description": "Preact preset for the vite bundler",
-	"type": "module",
-	"exports": "./dist/index.js",
+	"main": "./dist/cjs/index.js",
+	"module": "./dist/esm/index.mjs",
+	"exports": {
+		".": {
+			"import": "./dist/esm/index.mjs",
+			"require": "./dist/cjs/index.js"
+		},
+		"./package.json": "./package.json",
+		"./": "./"
+	},
 	"types": "dist/types/index.d.ts",
 	"scripts": {
 		"dev": "vite demo",
 		"dev:build": "vite build demo",
-		"build": "rimraf dist && tsc",
+		"build": "rimraf dist && tsc && tsc -p tsconfig.cjs.json && node tools/postbuild.mjs",
 		"prepublishOnly": "npm run build"
 	},
 	"keywords": [

--- a/tools/postbuild.mjs
+++ b/tools/postbuild.mjs
@@ -1,0 +1,14 @@
+import fs from "fs";
+import path from "path";
+import { fileURLToPath } from "url";
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+const dir = path.join(__dirname, "..", "dist", "esm");
+
+for (const file of fs.readdirSync(dir)) {
+	fs.renameSync(
+		path.join(dir, file),
+		path.join(dir, path.basename(file, path.extname(file)) + ".mjs"),
+	);
+}

--- a/tsconfig.cjs.json
+++ b/tsconfig.cjs.json
@@ -1,0 +1,9 @@
+{
+	"extends": "./tsconfig.json",
+	"compilerOptions": {
+		"outDir": "dist/cjs/",
+		"module": "CommonJS",
+		"declaration": true,
+		"declarationDir": "dist/types/"
+	}
+}

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,10 +1,10 @@
 {
-  "compilerOptions": {
-    "strict": true,
-    "esModuleInterop": true,
-    "module": "ESNext",
-    "moduleResolution": "Node",
-    "outDir": "dist"
-  },
-  "files": ["./src/index.ts"]
+	"compilerOptions": {
+		"strict": true,
+		"esModuleInterop": true,
+		"module": "ESNext",
+		"moduleResolution": "Node",
+		"outDir": "dist/esm/"
+	},
+	"files": ["./src/index.ts"]
 }

